core/store/src/trie/ops/resharding.rs

```diff
@@ -214,3 +254,113 @@ fn retain_decision(key: &[u8], intervals: &[Range<Vec<u8>>]) -> RetainDecision {
         RetainDecision::DiscardAll
     }
 }
+
+pub(crate) trait GenericTrieUpdateSplit<'a, N, V>:
```

Comment:
It was an explicit decision to name these Retain (and the name below would be RetainInner) because this is what is being done. We don't split shard, we only retain relevant keys, especially given that some ranges are left for both shards.

Reply:
Changed name to `Retain`